packages/core/src/awsService/ec2/sshKeyPair.ts

```diff
@@ -28,10 +29,25 @@ export class SshKeyPair {
     }
 
     public static async getSshKeyPair(keyPath: string, lifetime: number) {
+        SshKeyPair.assertValidKeypath(
```

Comment:
It looks like `SshKeyPair` constructor currently accepts a path controlled by the caller. If that option could be removed, so that `SshKeyPair` is fully in control, would that lead to a simpler design and possibly less need for validation? Just a thought

Reply:
+1. This looks like the values from `configureSshKeys()` could be set as defaults here.

Something like this, but `globals` may be undefined since it may be referenced before it is defined:
```
public static async getSshKeyPair(keyPath = path.join(globals.context.globalStorageUri.fsPath, `aws-ec2-key`), ...)
```

If you get issues, then you'll probably need to make it a function
```
public static async getSshKeyPair(getKeyPath = () => path.join(globals.context.globalStorageUri.fsPath, `aws-ec2-key`), ...)
```

Reply:
yeah, that makes sense. If the key class controls the key path then an illegal path isn't possible, avoiding the unnecessary checks.